### PR TITLE
[raz] Allow users with access to create own home dirs

### PR DESCRIPTION
**Pull request by Harshg999**

### CODE REVIEW

File: desktop/libs/aws/src/aws/s3/s3fs.py [hunk 1]
```diff
@@ -392,7 +392,24 @@ def filebrowser_action(self):
     return self._filebrowser_action
 
   def create_home_dir(self, home_path):
-    LOG.info('Create home directory is not available for S3 filesystem')
+    LOG.info('------------------>>>>>')
+    LOG.info(str(home_path))
+    LOG.info(str(self.user))
+    LOG.info(str(self.superuser))
+
+    # When S3 raz is enabled, try to create user dome dir for REMOTE_STORAGE_HOME path
```
Comment (JohanAhlen) [inline desktop/libs/aws/src/aws/s3/s3fs.py:400]: nit: dome -> home

**Review by JohanAhlen — APPROVED** [review]

**Comment by Harshg999** [comment]
The PR is ready for review. It will be squashed and merged once approved.

File: desktop/core/src/desktop/lib/fs/proxyfs.py [hunk 2]
```diff
@@ -200,9 +205,34 @@ def create(self, path, *args, **kwargs):
...
+    # Get the new home_path for S3/ABFS when RAZ is enabled.
+    if is_raz_s3():
+      home_path = get_s3_home_directory(User.objects.get(username=self.getuser()))
+    elif is_raz_abfs():
+      home_path = get_home_dir_for_abfs(User.objects.get(username=self.getuser()))
+
+    # Try getting user from the request and create home dirs. This helps when Hue admin is trying to create the dir for other users.
+    # That way only Hue admin needs authorization to create for all Hue users and not each individual user.
+    # If normal users also have authorization, then they can also create the dir for themselves if they want.
+    from crequest.middleware import CrequestMiddleware
```
Comment (JohanAhlen) [inline desktop/core/src/desktop/lib/fs/proxyfs.py:229]: import on top?

**Review by JohanAhlen — APPROVED** [review]
Nice!

**Pull request merged**